### DISCUSSION TITLE
Difference between getIPC() and getExecUsage

### DISCUSSION THREAD

**Discussion by arajeev9** [opcm/pcm#343] [Q&A]
I am trying to extract CPU counter information. I see 2 APIs:
**getIPC() description:** Computes average number of retired instructions per core cycle (IPC)
**getExecUsage() description:** Computes average number of retired instructions per time interval

Are the values correlated as  getExecUsage() = c * getIPC() ? Why do we have these 2 different APIs when it appears as if they are providing the same information ?

Thanks

**Comment by opcm** [comment]
IPC = instructions /cycles
ExecUsage = instructions / time

the number of cycles per time period is not constant because it depends on C-state and P-states.
